holmes/core/investigation.py

```diff
@@ -49,3 +57,92 @@ def investigate_issues(
         tool_calls=investigation.tool_calls or [],
         instructions=investigation.instructions,
     )
+
+
+def get_investigation_context(investigate_request: InvestigateRequest, dal: SupabaseDal, config: Config):
```

Comment:
Please consider to move it into `IssueInvestigator` Looks like we are duplicatin a lot of the code twice and it is maintanance hell. every change in issue investigate we will need to remember to fix twice as long as we have both implementations.

Reply:
We discussed this, its a tech debt and ill work on refactoring this function over the server and main so it will be more flexible